raiden/network/proxies/token_network.py

```diff
@@ -833,8 +876,18 @@ def unlock(
 
         leaves_packed = b''.join(lock.encoded for lock in merkle_tree_leaves)
 
+        gas_limit = self.proxy.estimate_gas(
+            'unlock',
+            channel_identifier,
+            self.node_address,
+            partner,
+            leaves_packed,
+        )
+        gas_limit = max(gas_limit or 0, UNLOCK_TX_GAS_LIMIT)
```

Comment:
since you use this a lot would it not be better to turn it into a function with a note of what you wrote [here](https://github.com/raiden-network/raiden/pull/2623/files#r221210617) to @palango explaining why you do it in the docstring?

```python
def get_gas_limit(given, precalculated):
    return max(given or 0, precalculated))
```

 Then at some point if we see estimate_gas is fixed upstream to never return None we can remove the function everywhere easily by grepping.

I am wondering when that happens? If the transaction is known to fail it should return the given `startGas` and not None.